metagraph/src/cli/parse_sequences.hpp

```diff
@@ -34,6 +34,12 @@ void parse_sequences(const std::string &file,
                      CallWeighted call_weighted_sequence) {
     mtg::common::logger->trace("Parsing '{}'", file);
 
+    if (config.graph_mode == graph::DeBruijnGraph::PRIMARY && file_format(file) != "FASTA") {
```

Comment:
What if the input is a KMC file with only canonical k-mers?

Reply:
In that case, the graph would have too many gaps, which definitely doesn't work for SuccinctDBG.
So I think it's better to require that a primary graph must be always constructed from primary contigs, so the input must be only FASTA.